src/java.base/share/native/libjimage/imageFile.cpp

```diff
@@ -331,7 +331,9 @@ bool ImageFileReader::open() {
     }
     // Memory map image (minimally the index.)
     _index_data = (u1*)osSupport::map_memory(_fd, _name, 0, (size_t)map_size());
-    assert(_index_data && "image file not memory mapped");
+    if (_index_data == nullptr) {
```

Comment:
The rest of the code in the `libjimage` library uses `NULL`, including the return value in `osSupport::map_memory(...)`. So I think it would be better to use `NULL` here for consistency.